backend/src/controllers/PUBLICTutorialController.ts

```diff
@@ -0,0 +1,114 @@
+import { Request, Response } from "express";
+import PUBLICTutorialService from "../services/PUBLICTutorialService";
+import { ObjectId, MongoError } from "mongodb";
+import { BAD_OBJECTID_PARSE_TEXT, INTERNAL_ERROR_TEXT, TUTORIAL_CATEGORIES } from "../constants";
+
+/**
+ * PUBLIC Tutorial route controller
+ *
+ * @author ale8k
+ */
+export default class PUBLICTutorialController {
+    /**
+     * Grabs a PUBLIC tutorial by its Id
+     *  - Endpoint takes the tutorial Id from the path params
+     *
+     * @async
+     * @param {Request} req the users request obj
+     * @param {Response} res our res obj
+     */
+    public static getTutorialById(req: Request, res: Response): void {
+        try {
+            const tutObjectId = new ObjectId(req.params.tutId);
+            PUBLICTutorialService.getPublicTutById("tutorials", tutObjectId).then(
+                (response) => {
+                    if (response === null) {
+                        res.sendStatus(204).end();
+                    } else {
+                        res.status(200).send(response).end();
+                    }
+                },
+                (err: MongoError) =>  res.status(500).send(INTERNAL_ERROR_TEXT + JSON.stringify(err)).end()
```

Comment:
can use `JSON.stringify(err, null, 2)` if err is an `object`;

Reply:
Idk if spaces necessary, stringify parses the object ordinarily anyway?

Reply:
gives a better format.

Reply:
But they'll parse it on other end no? Idk, seems futile.

Reply:
oh right. no problem. I thought we're giving the output on the console or something.